### ISSUE TITLE
About the accum_batch_size

### ISSUE THREAD

**Issue by Angzz** [sfzhang15/RefineDet#71]
@sfzhang15 Hi, what's the meaning of accum_batch_size, and its relation with iter_size? should I change this parameter for different numbers of GPU or different datasets?

Thanks!!!

**Comment by sfzhang15** [comment]
@Angzz 
The **batch_size** in the train.prototxt means how many images per GPU. The **iter_size** in the solver.prototxt means how many iterations to do a backpropagation. So the **accum_batch_size** = batch_size * num_gpu * iter_size,  which is the real batch size to do a  backpropagation.

**Comment by Angzz** [comment]
Thanks for your Reply!!!

Today I use ResNet101 for training (512X512 input size ), I train on one TITAN Xp(12GB), and the batch_size is only 2 available, this is much smaller than VGG16 with 12 per GPU, and the author of DSSD say if the batch_size per GPU below 5, it will harm the performance due to the batch norm layer, I want to ask whether it will affect RefineDet too, I know the memory consume is mainly because the ResNet101 is much deeper, and I also want to know what's the batch_size you use when train COCO with 512X512 inupt size?

**Comment by sfzhang15** [comment]
@Angzz 
For RefineDet_512_ResNet101, we train it using 4 M40 (24G) GPUs, every GPU has 5 images and the batch size is 20.

**Comment by Angzz** [comment]
Thanks a lot!!!

**Comment by miraclewkf** [comment]
@sfzhang15 You say `accum_batch_size = batch_size * num_gpu * iter_size`, but in `ResNet101_COCO_320.py` 
line 326 and line 327:
```batch_size_per_device = int(math.ceil(float(batch_size) / num_gpus))
 iter_size = int(math.ceil(float(accum_batch_size) / (batch_size_per_device * num_gpus)))```
these two lines mean 'accum_batch_size = batch_size * iter_size`

**Comment by sfzhang15** [comment]
@miraclewkf Hi,
My said `batch_size` is in the train.prototxt, i.e., [batch_size_per_device](https://github.com/sfzhang15/RefineDet/blob/master/examples/refinedet/ResNet101_COCO_320.py#L326) in the ResNet101_COCO_320.py
